src/Symfony/Component/Process/Process.php

```diff
@@ -297,15 +304,15 @@ public function start($callback = null)
 
         $this->process = proc_open($commandline, $descriptors, $this->processPipes->pipes, $this->cwd, $this->env, $this->options);
 
-        if ($ptsWorkaround) {
-            fclose($ptsWorkaround);
-        }
-
         if (!is_resource($this->process)) {
             throw new RuntimeException('Unable to launch a new process.');
         }
         $this->status = self::STATUS_STARTED;
 
+        if (isset($descriptors[3])) {
+            fclose($ptsWorkaround);
```

Comment:
moving this line looks wrong to me: you are not closing it anymore in case launching the process fails.